src/protocols/UBX.cpp

```diff
@@ -0,0 +1,131 @@
+#include <stdint.h>
+#include <stddef.h>
+
+#include <array>
+#include <vector>
+
+#include <gps_ublox/protocols/UBX.hpp>
+
+using namespace std;
+using namespace gps_ublox;
+using namespace protocols;
+
+const uint8_t UBX::SYNC_1 = 0xB5;
+const uint8_t UBX::SYNC_2 = 0x62;
+const uint8_t UBX::SYNC_1_IDX = 0;
+const uint8_t UBX::SYNC_2_IDX = 1;
+const uint8_t UBX::MSG_CLASS_IDX = 2;
+const uint8_t UBX::MSG_ID_IDX = 3;
+const uint8_t UBX::LENGTH_LSB_IDX = 4;
+const uint8_t UBX::LENGTH_MSB_IDX = 5;
+const uint8_t UBX::PAYLOAD_IDX = 6;
+const size_t UBX::FRAMING_SIZE_OVERHEAD = 8;
+
+UBX::Frame UBX::Frame::fromPacket(const uint8_t *buffer, size_t size)
+{
+    Frame frame;
+    frame.msg_class = buffer[MSG_CLASS_IDX];
+    frame.msg_id = buffer[MSG_ID_IDX];
+
+    uint16_t payload_size = buffer[LENGTH_LSB_IDX] | (buffer[LENGTH_MSB_IDX] << 8);
+    frame.payload = vector<uint8_t>(&buffer[PAYLOAD_IDX], &buffer[PAYLOAD_IDX] + payload_size);
+
+    return frame;
+}
+
+vector<uint8_t> UBX::Frame::toPacket() const
+{
+    vector<uint8_t> packet;
+    packet.push_back(SYNC_1);
+    packet.push_back(SYNC_2);
+    packet.push_back(this->msg_class);
```

Comment:
I personally really have a problem with the Python-style usage of `this->`. I'd rather not.

Reply:
We really should put together a style guide, for productivity sake. These are very personal style choices and it is virtually impossible to guess them right...

Reply:
Absolutely agreed. We should have a clang-formatter definition file and a validation step in vscode as well as in packages.

Let's have this scheduled once we get the motion box software ready. I also want to have the CI running, we can the next sprint with that.